core/domain/state_domain_test.py

```diff
@@ -458,13 +458,20 @@ def test_hints_validation(self):
         exploration.validate()
 
         # Add hint and delete hint.
-        init_state.add_hint(state_domain.SubtitledHtml('hint_2', 'new hint'))
+        init_state.interaction.hints.append(
```

Comment:
Use `.update_interaction_hints` instead. (see line 432)

Reply:
Thanks for the review @DubeySandeep 
Also is there any specific reason for this? Because `.update_interaction_hints` replaces the currently stored hints by the one passed to it and not append to it. To append I would have to make a separate list of hints to append the hints and then pass to `.update_interaction_hints`; which would make the same sense.
Or am I missing something?

Reply:
We use `update_interaction_hints` to update interaction hints, so from the top level, we should use this function to change the hints and test the expected value, that way we will ensure that updating hints functionality is working fine. I hope you understand!

Reply:
Ah Ok, now I get it.
Thanks!